tools/cross/format.py

```diff
@@ -69,28 +78,32 @@ def check_clang_format() -> bool:
         exit(1)
 
 
-def find_cpp_files(dir_path) -> List[str]:
+def find_files_by_exts(dir_path: str, exts: Tuple[str, ...]) -> List[str]:
     files = []
     for root, _, filenames in os.walk(dir_path):
         for filename in filenames:
-            if filename.endswith((".c++", ".h")):
+            if filename.endswith(exts):
                 files.append(os.path.join(root, filename))
     return files
 
 
-def clang_format(files: List[str], check: bool = False):
-    if not files:
-        logging.info("No changes to format")
-        exit(0)
+def clang_format(files: List[str], check: bool = False) -> bool:
     if check:
         result = subprocess.run(
-            [CLANG_FORMAT, "--verbose", "--dry-run", "--Werror"] + files, check=False
-        )
-        if result.returncode != 0:
-            logging.error("Code has lint. Fix with: python ./tools/cross/format.py")
-            exit(1)
+            [CLANG_FORMAT, "--verbose", "--dry-run", "--Werror"] + files)
+        return result.returncode == 0
     else:
-        subprocess.run([CLANG_FORMAT, "--verbose", "-i"] + files, check=False)
+        subprocess.run([CLANG_FORMAT, "--verbose", "-i"] + files)
+        return True
+
+
+def prettier(files: List[str], check: bool = False) -> bool:
```

Comment:
I'm not sure why this function is needed, or why we're running prettier from a py file? Running `pnpm exec prettier` from the repo root _should_ do everything we need—it shouldn't be necessary to explicitly list all the files?

Reply:
Ts script also handles clang-format, and will eventually take care of Python and Skylark (Bazel) formatting. @danlapid created this script to allow developers to reformat only files changed by their current commit. It's also set up as a git pre-commit hook.